.storybook/main.js

```diff
@@ -0,0 +1,38 @@
+const path = require('path');
+const { TsconfigPathsPlugin } = require('tsconfig-paths-webpack-plugin');
+
+/**
+ *  @callback StorybookWebpackConfig
+ *  @param {import("webpack").Configuration} config
+ *  @param {{configType: 'DEVELOPMENT' | 'PRODUCTION'}} options - change the build configuration. 'PRODUCTION' is used when building the static version of storybook.
+ *  @returns {import("webpack").Configuration}
+ */
+
+/**
+ *  @typedef {{check:boolean; checkOptions: Record<string,unknown>; reactDocgen: string | boolean; reactDocgenTypescriptOptions: Record<string,unknown>}} StorybookTsConfig
+ */
+
+/**
+ * @typedef {{stories: string[] ; addons: string[]; typescript: StorybookTsConfig; babel: (options:Record<string,unknown>)=>Promise<Record<string,unknown>>; webpackFinal: StorybookWebpackConfig}} StorybookConfig
+ */
+
+module.exports = /** @type {Pick<StorybookConfig,'addons' |'stories' |'webpackFinal'>} */ ({
+  stories: [],
+  addons: [
+    '@storybook/addon-essentials',
+    '@storybook/addon-a11y',
+    '@storybook/addon-knobs/preset',
+    'storybook-addon-performance',
```

Comment:
would we be OK reming the perf addon ? I have never seen it mentioned nor have I ever used it in any meaningful way

Reply:
we can do that, depends on requirements. It is used for react-examples that's why I added it here as well. 

My understanding was that this is gonna be used for re-render perf metrics in the future ?

Reply:
> re-render perf metrics

wasn't that why we create flamegrill ? 🤦‍♂️

Reply:
no idea